main.js

```diff
@@ -15,30 +13,18 @@ module.exports.loop = function () {
         }
     }
 
-    // Creating harvester automatically
     var harvesters = _.filter(Game.creeps, (creep) => creep.memory.role == 'harvester');
-    console.log('Harvesters: ' + harvesters.length);
-
-   
-    
-    if(harvesters.length < harvesterCount) {
-        var newName = 'Harvester' + Game.time;
-        console.log('Spawning new harvester: ' + newName);
-        Game.spawns['Spawn1'].spawnCreep([WORK,CARRY,MOVE], newName, 
-            {memory: {role: 'harvester'}});
-    }
-
-    // Creating upgrader automatically
     var upgraders = _.filter(Game.creeps, (creep) => creep.memory.role == 'upgrader');
-    console.log('Upgraders: ' +upgraders.length);
-
-    if(upgraders.length < upgraderCount) {
-        var newName = 'Upgrader' + Game.time;
-        console.log('Spawning new upgrader: ' + newName);
-        Game.spawns['Spawn1'].spawnCreep([WORK,CARRY,MOVE], newName, 
-            {memory: {role: 'upgrader'}});
+    
+    if(harvesters.length < harvesterCount && upgraders.length < upgraderCount) {
```

Comment:
I am skeptical of this. Consider OR "||" ?

Reply:
+1